document ModulesItertoolsUsageDeprecation (#7707)
fishtown-analytics/docs.getdbt.com: website/docs/reference/dbt-jinja-functions/modules.md

```

## itertools

:::info Note

> Starting with `dbt-core==1.10.6`, usage of `modules.itertools` will raise a deprecation warning. For more information and suggested workarounds, please refer to the [documentation on `ModulesItertoolsUsageDeprecation`](/reference/deprecations.md#modules-itertools-usage-deprecation).

Reviewer: ```suggestion
Starting in `dbt-core==1.10.6`, usage of `modules.itertools` raises a deprecation warning. For more information and suggested workarounds, refer to the [documentation on `ModulesItertoolsUsageDeprecation`](/reference/deprecations.md#modules-itertools-usage-deprecation).
```
